packages/medusa/src/services/discount.ts

```diff
@@ -211,6 +211,13 @@ class DiscountService extends TransactionBaseService {
         )) as Region[]
       }
 
+      if (isEmpty(discount.regions)) {
```

Comment:
todo: this check can be done first just before `const conditions = discount.rule?.conditions` since there is no need to do all that and then throw after since we can know before.
Also, instead of using external util to check the emptiness, I suggest using `if (!discount.regions?.length)`
Let me know what you think.